src/main/java/com/wolox/training/controllers/BookController.java

```diff
@@ -0,0 +1,67 @@
+package com.wolox.training.controllers;
+
+import com.wolox.training.dtos.BookDTO;
+import com.wolox.training.models.Book;
+import com.wolox.training.services.BookService;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping
+    public ResponseEntity<List<Book>> findAll() {
+        List<Book> books = bookService.findAll();
+        return new ResponseEntity<>(books, HttpStatus.OK);
+    }
+
+    @GetMapping("/book/{bookAuthor}")
+    public ResponseEntity<Book> findByAuthor(@PathVariable String bookAuthor) {
+        Optional<Book> bookOptional = bookService.findByAuthor(bookAuthor);
```

Comment:
You could simply do an .orElseThrow to the findByAuthor(bookAuthor) directly in the BookService, just as you do in other cases. And then just return the entity with the book here. It'll let you remove the if/else overhead.